lib/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+### lib/CMakeLists.txt ###
+#
+#
+
+## Platform and system libraries
+include(platform_libs.cmake)
+
+## imgui:
+execute_process(
+	COMMAND git submodule update --init imgui
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+add_library(imgui imgui/imgui.cpp imgui/imgui_draw.cpp)
+target_include_directories(imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/imgui")
+target_link_libraries(imgui ${PLATFORM_LIBS})
+target_compile_definitions(imgui PUBLIC "-DImDrawIdx=unsigned int") # short is not enough for us (long causes core dumps)
+set(LIBS ${LIBS} imgui)
+
```

Comment:
`imgui` has it's included directories added to the target on line 15. They are `PUBLIC` so targets that add `imgui` will inherit them automatically.

Re vars like `LIBS`, and `INCLUDE_DIRS`; I know they're not perfect - the idea was to keep library related stuff in `lib/` and just add a couple lists to the main target in `src/` without having to edit the main target whenever we add or drop a dependency.